raiden/network/transport/matrix/transport.py

```diff
@@ -1363,9 +1455,81 @@ def partner_joined(fetched_members: Optional[List[User]]) -> bool:
             )
 
             self._set_room_id_for_address(address, room.room_id)
-
             self.log.debug("Channel room", peer_address=to_checksum_address(address), room=room)
-            return room
+
+            # initiate web rtc handling
+            gevent.spawn(self.create_web_rtc_channel, address)
+
+    def _handle_sdp_callback(
+        self, rtc_session_description: Optional[RTCSessionDescription], partner_address: Address
+    ) -> None:
+        """
+        This is a callback function to process sdp (session description protocol) messages.
+        These messages are part of the ROAP (RTC Offer Answer Protocol) which is also called
+        signalling. Messages are exchanged via the partners' private matrix room.
+        Args:
+            rtc_session_description: sdp message for the partner
+            partner_address: Address of the partner
+        """
+        assert self._raiden_service is not None, "_raiden_service not set"
+        if rtc_session_description is None:
+            return
+
+        sdp_type = rtc_session_description.type
+
+        message = {"type": rtc_session_description.type, "sdp": rtc_session_description.sdp}
+        log.debug(
+            f"Send {sdp_type} to partner",
+            node=to_checksum_address(self._raiden_service.address),
+            partner_address=to_checksum_address(partner_address),
+            sdp_description=message,
+        )
+        room = self._get_room_for_address(partner_address, require_online_peer=True)
+        if room is None:
+            return
+
+        if sdp_type == "offer":
+            self._client.api.invite(room.room_id, message)
+        elif sdp_type == "answer":
+            self._client.api.answer(room.room_id, message)
+
+    def create_web_rtc_channel(self, partner_address: Address) -> None:
+        assert self._raiden_service is not None, "_raiden_service not set"
+
+        rtc_partner = self._web_rtc_manager.get_rtc_partner(partner_address)
+        log.debug(
+            "Waiting for initiating web rtc",
+            node=to_checksum_address(self._raiden_service.address),
+            partner_address=to_checksum_address(partner_address),
+        )
+        while (
+            self._get_room_for_address(partner_address, require_online_peer=True) is None
+            or not self._started
+        ):
+            if self._stop_event.is_set():
+                return
+            rtc_partner.partner_ready_event.wait(timeout=5)
```

Comment:
Is there a reason this is 5s? Can we use a general timeout here?

Reply:
I used it from a similar approach in the codebase. Do you mean using a constant variable instead of inline integer or an already existing one?

Reply:
Yeah, I just thought we had a constant for such timeouts, but doesn't seem like it, so ignore please.